ibis/backends/sqlite/__init__.py

```diff
@@ -95,169 +76,518 @@ def do_connect(
+    def begin(self):
+        cur = self.con.cursor()
+        try:
+            yield cur
+        except Exception:
+            self.con.rollback()
+            raise
+        else:
+            self.con.commit()
+        finally:
+            cur.close()
 
-    @staticmethod
-    def _new_sa_metadata():
-        meta = sa.MetaData()
-
-        @sa.event.listens_for(meta, "column_reflect")
-        def column_reflect(inspector, table, column_info):
-            if type(column_info["type"]) is TIMESTAMP:
-                column_info["type"] = ISODATETIME()
+    def list_databases(self, like: str | None = None) -> list[str]:
+        with self._safe_raw_sql("SELECT name FROM pragma_database_list()") as cur:
+            results = [r[0] for r in cur.fetchall()]
 
-        return meta
+        return sorted(self._filter_with_like(results, like))
 
-    def _table_from_schema(
-        self, name, schema, database: str | None = None, temp: bool = True
-    ) -> sa.Table:
-        prefixes = []
-        if temp:
-            prefixes.append("TEMPORARY")
-        columns = self._columns_from_schema(name, schema)
-        return sa.Table(
-            name, sa.MetaData(), *columns, schema=database, prefixes=prefixes
+    def list_tables(
+        self,
+        like: str | None = None,
+        database: str | None = None,
+    ) -> list[str]:
+        if database is None:
+            database = "main"
+
+        sql = (
+            sg.select("name")
+            .from_(F.pragma_table_list())
+            .where(
+                C.schema.eq(database),
+                C.type.isin("table", "view"),
+                ~(
+                    C.name.isin(
+                        "sqlite_schema",
+                        "sqlite_master",
+                        "sqlite_temp_schema",
+                        "sqlite_temp_master",
+                    )
+                ),
+            )
+            .sql(self.name)
         )
+        with self._safe_raw_sql(sql) as cur:
+            results = [r[0] for r in cur.fetchall()]
 
-    @property
-    def _current_schema(self) -> str | None:
-        return self.current_database
+        return sorted(self._filter_with_like(results, like))
 
-    def _metadata(self, query: str) -> Iterator[tuple[str, dt.DataType]]:
-        view = f"__ibis_sqlite_metadata_{util.guid()}"
+    def _parse_type(self, typ: str, nullable: bool) -> dt.DataType:
+        typ = typ.lower()
+        try:
+            out = self._type_map[typ]
+        except KeyError:
+            return self.compiler.type_mapper.from_string(typ, nullable=nullable)
+        else:
+            return out.copy(nullable=nullable)
+
+    def _inspect_schema(
+        self, cur: sqlite3.Cursor, table_name: str, database: str | None = None
+    ) -> Iterator[tuple[str, dt.DataType]]:
+        if database is None:
+            database = "main"
+
+        sql = f'SELECT name, type, "notnull" FROM "{database}".pragma_table_info("{table_name}")'
```

Comment:
It might be preferable to use sqlglot queries so ensure that the quoting/escaping is right.

Reply:
AFAICT there's no way in sqlglot to namespace a function call like this (like `{possibly_quoted_namespace}.{func_name}(...)`).

I've updated the code to use sqlglot to manually quote the identifiers as needed everywhere string templating was used.